backend/src/common/util/parameter_formatter.go

```diff
@@ -42,8 +42,18 @@ type ParameterFormatter struct {
 	index          int64
 }
 
-// NewParameterFormatter returns a new ParameterFormatter.
-func NewParameterFormatter(runUUID string, scheduledEpoch int64, nowEpoch int64,
+// NewRunParameterFormatter returns a new ParameterFormatter to substitute run macros.
+func NewRunParameterFormatter(runUUID string, runAt int64) *ParameterFormatter {
+	return &ParameterFormatter{
+		runUUID:        runUUID,
+		nowEpoch:       runAt,
+		scheduledEpoch: -1, // invalidate field
```

Comment:
What does the comment mean?

Reply:
@Bobgy it's a way to "disable" that struct field. See here https://github.com/kubeflow/pipelines/pull/5086/files/ad8621f9295f9ecbfc942de66c782b2d36a3c435#diff-8e2dc598b0c65a45c7c52bb76157cced997b3b0bc9d7c23f95395737cc8c72baR98 Since in one-off runs we don't care about `scheduledEpoch` or `index`. If you agree on the approach I can change the comment to `Disable field, see createSubstitutes function`.

Reply:
If `0` can be used as an empty field, that would be preferable. I think we can use 0 for epoch related fields.

If not, I'd suggest wrapping it with a pointer, so it's possible to represent `nil`. I don't like using magic numbers like `-1` to indicate emptiness.

@capri-xiyue what do you think?

Reply:
In some language, 
negative epoch means dates below '1970-01-01 00:00:00'. 
I'd prefer we represent it using`nil`

Reply:
@capri-xiyue Maybe I am missing something, but since `runUUID`, `newEpoch`, `scheduledEpoch`, and  `index` are of type `string` or `int64` we cannot assign `nil`, right?

Reply:
@capri-xiyue any suggestion? I would like to tackle this in these days so that we can get it merged in time for Kubeflow 1.3

Reply:
Hi @StefanoFioravanzo,

The original suggestion is to change the type to a pointer: e.g. `*int` or `*string`, so that it's possible to use `nil` as empty value.

Anyway, I'm not very concerned with using `-1`.
But if we decide to go with `-1`, I'd suggest creating a const for it, so it's not a magic number.

Reply:
@Bobgy you are right you had proposed it but I had missed your comment, my bad. I just added a commit to move -1 to a constant. I would look into making the types be pointers, but I can't allocate much time this week so if it's fine for you to keep the constant let's go with it

Reply:
And I don't get why the build suddenly fails, especially in a code path that I didn't touch with the new commit 🤔